tests/core/daemon/test_daemon_register.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import annotations
+
+import aiohttp
+import pytest
+
+from chia.daemon.server import WebSocketServer
+from chia.simulator.block_tools import BlockTools
+from chia.util.ws_message import create_payload
+
+
+@pytest.mark.asyncio
+async def test_multiple_register_same(get_daemon: WebSocketServer, bt: BlockTools) -> None:
+    ws_server = get_daemon
+    config = bt.config
+
+    daemon_port = config["daemon_port"]
+
+    # setup receive service to connect to the daemon
+    async with aiohttp.ClientSession() as client:
+        ws = await client.ws_connect(
+            f"wss://127.0.0.1:{daemon_port}",
+            autoclose=True,
+            autoping=True,
+            ssl_context=bt.get_daemon_ssl_context(),
+            max_msg_size=100 * 1024 * 1024,
+        )
+
+        service_name = "test_service"
+        data = {"service": service_name}
+        payload = create_payload("register_service", data, service_name, "daemon")
+        for _ in range(4):
+            await ws.send_str(payload)
+            await ws.receive()
```

Comment:
Verify indication of success?

Reply:
I don't think it's needed, the test will fail if this didn't work because the connections won't match in the next line or so and it would only be slightly more helpful to fail earlier.

Reply:
Yes, the idea is clarity of the failure.  That it isn't that the server didn't track the connection but that you didn't even talk to the server successfully.